packages/ui-extensions-sdk/src/utils/testUtils.ts

```diff
@@ -123,7 +123,9 @@ export class MockFramePostChildClient {
         );
     }
 
-    request(eventType: string, data: any): any {}
+    request(eventType: string, data: any): any {
+        return this.mockRequest(eventType, data);
+    }
```

Comment:
Not sure if this was intentionally omitted, but by mocking the request automatically when `framepostClient.request` is called, we can test client methods that rely on the top-level `DDClient` classes `request` method.

Reply:
I think conceptually, it's not supposed to work, but it does because we're being lax with our testing.

If we think about the two sides of the system, we've got the Datadog platform and the SDK. The way things are supposed to work is that one side uses `request` and the other side uses `onRequest`. E.g. if the SDK uses `onRequest(RequestType.GET_WIDGET_CONTEXT_MENU_ITEMS, …)`, it's expected that the platform is going to use `request(RequestType.GET_WIDGET_CONTEXT_MENU_ITEMS, …)`; if the SDK uses `request(RequestType.SEND_NOTIFICATION, …)`, it's expected that the platform is going to use `onRequest(RequestType.SEND_NOTIFICATION, …)`.

What's not expected is that one side both sends a request and also responds to the same request. I'm not even sure it would actually do anything if you did that because the request is never routed to the sender always to the receiver.

Long story short, I think this is the reason the `request` method here had no implementation, because we haven't actually tested that when we end up calling `request` that it makes it across to the other side and returns something (because there is no other side in these tests).

I think what we do in other tests that make requests is setup the mock:
- https://github.com/DataDog/apps/blob/91301a66a276f39090db1f14c263e6a966e5edd9/packages/ui-extensions-sdk/src/notification/notification.test.ts#L23-L25
- https://github.com/DataDog/apps/blob/91301a66a276f39090db1f14c263e6a966e5edd9/packages/ui-extensions-sdk/src/api/api.test.ts#L13-L15
and validate that the right stuff is sent:
- https://github.com/DataDog/apps/blob/91301a66a276f39090db1f14c263e6a966e5edd9/packages/ui-extensions-sdk/src/notification/notification.test.ts#L32-L38
- https://github.com/DataDog/apps/blob/91301a66a276f39090db1f14c263e6a966e5edd9/packages/ui-extensions-sdk/src/api/api.test.ts#L26-L37

We don't seem to validate that the right stuff is returned.

Reply:
Ahh I see. I think it makes more sense to validate what we're passing to the underlying methods rather than checking what's being returned. I'll update.